sdk/dotnet/Pulumi/Resources/StackReference.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright 2016-2019, Pulumi Corporation
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Pulumi.Serialization;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// Manages a reference to a Pulumi stack and provides access to the referenced stack's outputs.
+    /// </summary>
+    public class StackReference : CustomResource
+    {
+        /// <summary>
+        /// The name of the referenced stack.
+        /// </summary>
+        [Output("name")]
+        public Output<string> Name { get; private set; } = null!;
+
+        /// <summary>
+        /// The outputs of the referenced stack.
+        /// </summary>
+        [Output("outputs")]
+        public Output<ImmutableDictionary<string, object>> Outputs { get; private set; } = null!;
+
+        /// <summary>
+        /// The names of any stack outputs which contain secrets.
+        /// </summary>
+        [Output("secretOutputNames")]
+        public Output<ImmutableArray<string>> SecretOutputNames { get; private set; } = null!;
```

Comment:
can probably be internal.  i don't think we want to actually expose this to user (same for 'Outputs').

Reply:
changed

Reply:
Had to revert this because we ignore non-public properties: https://github.com/pulumi/pulumi/blob/master/sdk/dotnet/Pulumi/Serialization/OutputCompletionSource.cs#L61